packages/runed/src/lib/utilities/element-size/element-size.svelte.ts

```diff
@@ -79,14 +79,56 @@ export class ElementSize {
 		this.#node = node;
 
 		this.#size = {
-			width: options.initialSize?.width ?? 0,
-			height: options.initialSize?.height ?? 0,
+			width: 0,
+			height: 0,
 		};
 	}
 
+	calculateSize() {
+		const element = get(this.#node);
+
+		// no element or no window, return undefined, we will return the initial size
+		if (!element || !this.#window) {
+			return;
+		}
+
+		const offsetWidth = element.offsetWidth;
+		const offsetHeight = element.offsetHeight;
+
+		// easy mode, just return offsets
+		if (this.#options.box === "border-box") {
+			return {
+				width: offsetWidth,
+				height: offsetHeight,
+			};
+		}
+
+		// hard mode, we need to calculate the content size
+		const style = this.#window.getComputedStyle(element);
```

Comment:
can we not use `getBoundingClientRect` for this?

Reply:
Isn't `getBoundingClientRect` slower? What advantages would have?

Reply:
Oh i misread where the comment was...no unfortunately `getBoundingClientRect` only returns the bounding box which is the border box so we still need to get the computed style for content-box

Reply:
aha you are right. i missed that

ignore me :D